commit cfa074bdc5fb6520874422ded0cd39b3a110d7fc
Author: Tang Yiwei <tangyiwei.2022@bytedance.com>
Date:   Thu Nov 17 10:28:53 2022 +0800

    nbd-proxy:  Disable nbd-client's netlink support
    
    When openbmc/openbmc updated the nbd package from 3.20 to 3.24 [1], it
    also added libnl as a dependency, enabling nbd's HAVE_NETLINK feature.
    When using netlink instead of the older ioctl interface, the nbd-client
    process exits immediately after setting up the connection, which
    confuses nbd-proxy.  A more thorough, long-term solution would be to
    restructure nbd-proxy around the newer nbd-client netlink behavior
    (nbd-client's support for the ioctl interface is slated to be removed at
    some unspecified point in the future [2]), but as a simpler fix for the
    time being we can just disable netlink support via nbd-client's -L
    command-line flag.
    
    [1] https://github.com/openbmc/openbmc/commit/78b727985e7571e0b196561e44427690f04d57d9#diff-452e7c988a0afa8d4d41dc2c7710a379997ef7c6d37def5b588bd7593ae694c0
    [2] https://github.com/NetworkBlockDevice/nbd/blob/90dc1adc9ba196fca84a9b4f7d85fdfdd9fc4d5b/man/nbd-client.8.in.sgml#L258-L260
    
    Signed-off-by: Tang Yiwei <tangyiwei.2022@bytedance.com>
    Signed-off-by: Zev Weiss <zev@bewilderbeest.net>
    Signed-off-by: Jian Zhang <zhangjian.3032@bytedance.com>
    Change-Id: I29a76af0f1150aef1f42b54686b45c72848f43da

diff --git a/nbd-proxy.c b/nbd-proxy.c
index bffe1f9..54c8cf7 100644
--- a/nbd-proxy.c
+++ b/nbd-proxy.c
@@ -156,6 +156,7 @@ static int start_nbd_client(struct ctx *ctx)
 		execlp("nbd-client", "nbd-client",
 				"-u", ctx->sock_path,
 				"-n",
+				"-L",
 				"-t", timeout_str,
 				ctx->config->nbd_device,
 				NULL);
